src/h3lib/lib/algos.c

```diff
@@ -411,6 +411,32 @@ H3Index h3NeighborRotations(H3Index origin, Direction dir, int* rotations) {
     return out;
 }
 
+/**
+ * Get the direction for a given neighbor. This is effectively the reverse
+ * operation for h3NeighborRotations. Returns INVALID_DIGIT if the cells are
+ * not neighbors.
+ *
+ * TODO: This is currently a brute-force algorithm, but as it's O(6) that's
+ * probably acceptible.
+ */
+Direction directionForNeighbor(H3Index origin, H3Index destination) {
```

Comment:
I broke this out from `getH3UnidirectionalEdge` for reuse. I tried several much more clever approaches to getting the direction, but they all ended up being significantly slower than just checking all neighbors 🤷‍♂️